src/ResourceManager/Websites/Commands.Websites/Cmdlets/AppServicePlans/SetAzureAppServicePlan.cs

```diff
@@ -58,7 +58,7 @@ public override void ExecuteCmdlet()
                     AppServicePlan = WebsitesClient.GetAppServicePlan(ResourceGroupName, Name);
                     AppServicePlan.Sku.Tier = string.IsNullOrWhiteSpace(Tier) ? AppServicePlan.Sku.Tier : Tier;
                     AppServicePlan.Sku.Capacity = NumberofWorkers > 0 ? NumberofWorkers : AppServicePlan.Sku.Capacity;
-                    var workerSizeAsNumber = int.Parse(AppServicePlan.Sku.Name.Substring(1, AppServicePlan.Sku.Name.Length - 1));
+                    var workerSizeAsNumber = int.Parse(Regex.Match(AppServicePlan.Sku.Name, @"\d+").Value);
```

Comment:
To avoid issues in the future, what if no digit value is found, seems like there should be a default value in this case.

Also, can you add some regression automation for this scenario?  If you have verified locally, you can file an issue for test automation in a near term sprint.  Please either add the test automation or link the test issue here.